lambda/lambda.ml

```diff
@@ -2071,8 +2072,8 @@ let primitive_may_allocate : primitive -> locality_mode option = function
   | Pbox_float (_, m) | Pbox_int (_, m) | Pbox_vector (_, m) -> Some m
   | Prunstack | Presume | Pperform | Preperform
     (* CR mshinwell: check *)
-  | Ppoll ->
-    Some alloc_heap
+  | Ppoll -> Some alloc_heap
+  | Pcpu_relax -> if Config.poll_insertion then None else Some alloc_heap
```

Comment:
Certainly `Pcpu_relax` should match `Ppoll` in the non-poll-insertion case, but I don't understand why either of them is marked as `Some alloc_heap`. What's this function used for?

Reply:
It appears to only be used for checking whether a primitive stack-allocates, so the heap cases don't matter. There's also a CR @mshinwell saying to check this case, so it could probably be removed (in another pr)